services/ext/api.go

```diff
@@ -1086,6 +1086,14 @@ func (api *PublicAPI) UnreadAndAcceptedActivityCenterNotificationsCount() (uint6
 	return api.service.messenger.UnreadAndAcceptedActivityCenterNotificationsCount()
 }
 
+func (api *PublicAPI) HasUnseenActivityCenterNotifications() (bool, error) {
```

Comment:
@MishkaRogachev, @cammellos, what are your thoughts on my previous comment? Downsides/upsides? I generally prefer if we could write clients that don't need to constantly pull data from `status-go`, except for an initial data load.

Reply:
@ilmotta I temporary switched for an urgent issue in status-desktop and i need to finish some experiment so that I can form an opinion. Initially I've planed to use existing signal about new notification and call `HasUnseenActivityCenterNotifications` only when activity center popup is closed

Reply:
Totally understandable :mechanical_arm:

Reply:
@ilmotta i've tested it [here](https://github.com/status-im/status-desktop/pull/9479) and i think there is no need for an additional signal, at least for desktop. I call `unreadActivityCenterNotificationsCount` and `hasUnseenActivityCenterNotifications` on start and when new notifications arrive. When i close activity center popup i also call `markActivityCenterNotificationRead` and `hasUnseenActivityCenterNotifications` to double check state saved.

Reply:
so, the way I would naturally code it on status-react is:

1) Have and endpoint that returns the has-seen state (as per PR)
2) Everytime a notification arrives as a signal, the client will check 
```
is the user on the activity center?
 no -> mark as seen state to false in the client state 
 yes -> don't update ui, mark as seen to true
```

The backend cannot fully drive this piece of state since it's contextual to where the user is.

There's some complications though, for example notifications can also be removed from the network, in that case, `hasSeen` should disappear. 
the case for example would be something like (there's many more though):

1) User A receives a contact request from user B, is not on the activity center, has_seen is true
2) User B retracts the contact request, the notification is removed from user A device. has_seen should be false.

In this case the frontend will need a signal to check whether this has changed (or of course poll, but we tend not to poll in status-react).

Whether we want this to be a separate piece of state (MessengerResponse.ActivityCenterHasSeen say), or a derivation of notifications state (not sure that works, seems harder), it's something we could discuss, but my 2p is that at some point, we will need a signal (or polling)

Reply:
`User B retracts the contact request, the notification is removed from user A device. has_seen should be false.`
In this case notification gets updated with `read == false` and it should work in same way as regular notification. I'll check it tomorrow

Reply:
So just to confirm @MishkaRogachev, you think we won't need signals at all? Or do you think a follow up PR might be necessary to implement that?

Reply:
@ilmotta `ActivityCenterState` is added to response. For sync endpoints those responses are delivered as return value from rpc, but for async they delivered via singal. @cammellos does i get it right?